examples/demo/modules/blog/cms_blog_api.e

```diff
@@ -8,93 +8,108 @@ class
-			-- Number of nodes of type blog
+			-- Number of nodes of type blog.
 		do
-			Result := node_storage.blogs_count
+			Result := blog_storage.blogs_count
 		end
 
-	blogs_count_from_user (user_id: INTEGER_64) : INTEGER_64
-			-- Number of nodes of type blog from user with user_id
+	blogs_count_from_user (a_user: CMS_USER): INTEGER_64
+			-- Number of nodes of type blog from user with `a_user_id'.
+		require
+			has_id: a_user.has_id
 		do
-			Result := node_storage.blogs_count_from_user(user_id)
+			Result := blog_storage.blogs_count_from_user (a_user)
 		end
 
-	blogs_order_created_desc: LIST[CMS_NODE]
+	blogs_order_created_desc: LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date (descending)
 		do
-			Result := add_authors(node_storage.blogs)
+			Result := nodes_to_blogs (blog_storage.blogs)
 		end
 
-	blogs_order_created_desc_limited (a_limit:NATURAL_32; a_offset:NATURAL_32) : LIST[CMS_NODE]
+	blogs_order_created_desc_limited (a_limit: NATURAL_32; a_offset: NATURAL_32): LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date and limited by limit and offset
-		local
-			tmp: LIST[CMS_NODE]
 		do
-			-- load all posts and add the authors to each post
-			Result := add_authors(node_storage.blogs_limited (a_limit, a_offset))
-
+				-- load all posts and add the authors to each post
+			Result := nodes_to_blogs (blog_storage.blogs_limited (a_limit, a_offset))
 		end
 
-	blogs_from_user_order_created_desc_limited (a_user_id: INTEGER_32; a_limit:NATURAL_32; a_offset:NATURAL_32) : LIST[CMS_NODE]
+	blogs_from_user_order_created_desc_limited (a_user: CMS_USER; a_limit: NATURAL_32; a_offset: NATURAL_32) : LIST [CMS_BLOG]
 			-- List of nodes ordered by creation date and limited by limit and offset
-		local
-			tmp: LIST[CMS_NODE]
+		require
+			has_id: a_user.has_id
 		do
-			-- load all posts and add the authors to each post
-			Result := add_authors(node_storage.blogs_from_user_limited (a_user_id, a_limit, a_offset))
-
+				-- load all posts and add the authors to each post
+			Result := nodes_to_blogs (blog_storage.blogs_from_user_limited (a_user, a_limit, a_offset))
 		end
 
 feature {NONE} -- Helpers
 
-	add_authors(posts: LIST[CMS_NODE]) : LIST[CMS_NODE]
-			-- sets the authors of all given sql results	
+	nodes_to_blogs (a_nodes: LIST [CMS_NODE]): ARRAYED_LIST [CMS_BLOG]
+			-- Convert list of nodes into a list of blog when possible.
 		do
-			Result := posts
-			if posts /= Void then
+			create {ARRAYED_LIST [CMS_BLOG]} Result.make (a_nodes.count)
+
+			if attached node_api as l_node_api then
 				across
-					Result
-				as
-					sql_result
+					a_nodes as ic
 				loop
-					if
-						sql_result.item /= Void and then
-						attached {CMS_PARTIAL_USER} sql_result.item.author as l_partial_author
-					then
-						if attached cms_api.user_api.user_by_id (l_partial_author.id) as l_author then
-							sql_result.item.set_author (l_author)
-						end
+					if attached {CMS_BLOG} l_node_api.full_node (ic.item) as l_blog then
```

Comment:
here we reuse the node api implementation to fill author .. and other potential partial information.
at some point, we will improve the node api and storage to offer more function, and I guess all the SQL statement used currently for Blog will be away.